### ISSUE TITLE
TFIBDatabase.AttachmentID некорректный после восстановления коннекта

### ISSUE THREAD

**Issue by bvsoft** [madorin/fibplus#34]
Приветствую!
Свойство TFIBDatabase.AttachmentID не проинициализируется заново, если случился разрыв коннекта у базы данных, и мы заново его восстанавливаем (ведь TFibDatabase.close не выполнился, а инициализация vAttachmentID происходит либо в Create, либо в Close). Исправится, если  в TFibDatabase.Open перед обращением к свойству AttachmentID вставить инициализацию переменной vAttachmentID:

```
  ...
  vInternalTransaction.Timeout      :=1000;  
  FStreammedConnectFail:=False;
  vAttachmentID    :=-1; //<--- here
  AttachmentID;
  DPB:=FDPB;
  ...
```

**Comment by walxo** [comment]
Можно показать более подробно, где и как это должно быть размещено.

**Comment by bvsoft** [comment]
Модуль FibDatabase.pas, функция procedure TFIBDatabase.Open(RaiseExcept:boolean = True);
строка 2050

**Comment by walxo** [comment]
Спасибо!

**Comment by madorin** [comment]
@bvsoft , @walxo , наверное лучше сделать сброс vAttachmentID в DoOnLostConnect ?
Здесь: https://github.com/madorin/fibplus/blob/cad23f81a9e6994fdac502c1857625efa2700991/pFIBDatabase.pas#L571
В этом случае AttachmentID будет невалидный и до вызова Open, сразу после потеря соединения

**Comment by madorin** [comment]
Сделал изменение в этом брэнче, можете протестировать
https://github.com/madorin/fibplus/compare/34-attachement-id

**Comment by VladimirBorzov** [comment]
Да, конечно, здесь тоже обязательно. Лучше и там, и там оставить...
